src/charm.py

```diff
@@ -175,29 +217,160 @@ def _generate_cu_config(self) -> str:
                 "DU F1 port information not available. Using default value %s", DU_F1_DEFAULT_PORT
             )
             du_f1_port = DU_F1_DEFAULT_PORT
-        if not (pod_ip := _get_pod_ip()):
-            logger.warning("Pod IP address not available")
+        if not (
+            self._charm_config.f1_ip_address
+            and self._charm_config.n2_ip_address
+            and self._charm_config.n3_ip_address
+        ):
+            logger.warning("Interfaces ip addresses are not available")
             return ""
         if not self._n2_requirer.amf_ip_address:
             logger.warning("AMF IP address not available")
             return ""
         return _render_config_file(
             gnb_name=self._gnb_name,
             cu_f1_interface_name=self._charm_config.f1_interface_name,
-            cu_f1_ip_address=pod_ip,
+            cu_f1_ip_address=str(self._charm_config.f1_ip_address).split("/")[0],
             cu_f1_port=self._charm_config.f1_port,
             du_f1_port=du_f1_port,
             cu_n2_interface_name=self._charm_config.n2_interface_name,
-            cu_n2_ip_address=pod_ip,
+            cu_n2_ip_address=str(self._charm_config.n2_ip_address).split("/")[0],
             cu_n3_interface_name=self._charm_config.n3_interface_name,
-            cu_n3_ip_address=pod_ip,
+            cu_n3_ip_address=str(self._charm_config.n3_ip_address).split("/")[0],
             amf_external_address=self._n2_requirer.amf_ip_address,
             mcc=self._charm_config.mcc,
             mnc=self._charm_config.mnc,
             sst=self._charm_config.sst,
             tac=self._charm_config.tac,
         )
 
+    def _generate_network_annotations(self) -> List[NetworkAnnotation]:
+        """Generate a list of NetworkAnnotations to be used by CU's StatefulSet.
+
+        Returns:
+            List[NetworkAnnotation]: List of NetworkAnnotations
+        """
+        return [
+            NetworkAnnotation(
+                name=f"{self._charm_config.n3_interface_name}-net",
+                interface=self._charm_config.n3_interface_name,
+            ),
+            NetworkAnnotation(
+                name=f"{self._charm_config.n2_interface_name}-net",
+                interface=self._charm_config.n2_interface_name,
+            ),
+            NetworkAnnotation(
+                name=f"{self._charm_config.f1_interface_name}-net",
+                interface=self._charm_config.f1_interface_name,
+            ),
+        ]
+
+    def _get_n3_nad_config(self) -> dict:
+        n3_nad_config = {
+            "cniVersion": "0.3.1",
+            "ipam": {
+                "type": "static",
+                "addresses": [
+                    {
+                        "address": self._charm_config.n3_ip_address,
+                    }
+                ],
+            },
+            "capabilities": {"mac": True},
+        }
```

Comment:
We used to define a `base_config` to avoid duplicating this piece of code in every NAD config. Is there any reason for not doing that anymore?

Reply:
Base config is used again.